runtime/lua/vim/lsp.lua

```diff
@@ -651,6 +665,14 @@ function lsp.start_client(config)
           pcall(handlers.on_error, lsp.client_errors.ON_INIT_CALLBACK_ERROR, err)
         end
       end
+      if config.settings and send_settings then
+        if vim.tbl_isempty(config.settings) then
+          config.settings = {[vim.type_idx]=vim.types.dictionary}
```

Comment:
```suggestion
          config.settings = vim.empty_dict()  -- ensure it's sent as dict, not as array.
```

I think this way the intention/reason for this would be a bit clearer - but probably personal preference, so feel free to ignore.

Reply:
Done, I agree.